Add note about @ character in logs search (#28506)
DataDog/documentation: content/en/logs/explorer/search_syntax.md

## Escape special characters and spaces

The following characters, which are considered special: `+` `-` `=` `&&` `||` `>` `<` `!` `(` `)` `{` `}` `[` `]` `^` `"` `“` `”` `~` `*` `?` `:` `\` `#`, and spaces require escaping with the `\` character. 
- `/` is not considered a special character and doesn't need to be escaped.

> - `@` currently is not searchable in the logs explorer as it is reserved for [Attribute Search](#attributes-search).

Reviewer: ```suggestion
- `@` cannot be used in search queries within Logs Explorer because it is reserved for [Attribute Search](#attributes-search).
```
